spec.bs

```diff
@@ -221,6 +224,35 @@ The <dfn for="interest group">estimated size</dfn> of an [=interest group=] |ig|
     [=interest group ad/render url=].
   1. The [=string/length=] of |ad|'s [=interest group ad/metadata=] if the field is not null.
 
+<div algorithm="check interest group permissions">
+
+To <dfn>check interest group permissions</dfn> given a [=string=] <dfn
+for="fetch interest group permissions">|ownerOrigin|</dfn>, a [=string=] <dfn
+for="fetch interest group permissions">|frameOrigin|</dfn>, and a [=boolean=]
+<dfn for="fetch interest group permissions">|isJoin|</dfn>:
+1. If |ownerOrigin| is equal to |frameOrigin|, then return true.
+1. Let |permissionsUrl| be the result of [building an interest group permissions url=]
+  with |ownerOrigin| and |frameOrigin|.
+1. Let |resource| be the result of [=fetching resource=] with |permissionsUrl|

```

Comment:
The "fetching resource" synchronously waits for fetching to complete which we definitely shouldn't be doing on the main thread here. I think we'd probably want to make a normal fetch and set "the rest of these steps" to the Fetch's process response body algorithm: https://fetch.spec.whatwg.org/#process-response-end-of-body or so.

The good news is that gives you the opportunity to create your own request and make its request mode CORS, removing the TODO below (assuming that's what implementations do today).

Reply:
@domfarolino As I said in the other comment for line 177, does it make sense to call the permission check in the parallel queue to run it in parallel, then the "fetching resource" is no longer blocking the main thread?